qiskit_neko/tests/machine_learning/test_neural_network_classifier.py

```diff
@@ -14,7 +14,6 @@
 import numpy as np
 from ddt import ddt, data
 
-from qiskit.primitives import Sampler as ReferenceSampler
```

Comment:
`qiskit-machine-learning` actually supports V2 primitives, I haven't tested it but in principle this could be the only change you'd have to make in this file:

`from qiskit.primitives import StatevectorSampler as ReferenceSampler`

Should we give it a try?